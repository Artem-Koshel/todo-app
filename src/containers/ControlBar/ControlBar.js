import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { setSelectedFilter } from '../../redux/slices/filterSlice';
import TextRadioButtons from '../../components/TextRadioButtons/TextRadioButtons';
import styles from './ControlBar.module.scss';
import { removeItems } from '../../redux/slices/todoListActions';

export default function ControlBar() {
  const selectedFilter = useSelector((state) => state.selectedFilter);
  const items = useSelector((state) => state.todoList);
  const itemsLeft = items.filter((item) => !item.done).length;
  const dispatch = useDispatch();

  const handleClearCompleted = useCallback(() => {
    const idsToRemove = items.filter((item) => item.done).map((item) => item.id);
    dispatch(removeItems(idsToRemove));
  });

  return (
    <div className={styles.controlBar}>
      <span className={styles.filters}>{`${itemsLeft} items left`}</span>
      <TextRadioButtons
        name="filters"
        selectedValue={selectedFilter}
        values={['All', 'Active', 'Completed']}
        onChange={(value) => dispatch(setSelectedFilter(value))}
      />
      <button className={styles.clearCompleted} type="button" onClick={handleClearCompleted}>
        Clear completed
      </button>
    </div>
  );
}
