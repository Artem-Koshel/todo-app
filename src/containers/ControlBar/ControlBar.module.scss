.controlBar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 10px 30px;
    font-size: small;
    color: gray;
    height: 25px;
}

.clearCompleted {
    border: none;
    background: none;
    padding: 0;
    font-size: small;
    color: gray;
    cursor: pointer;
    font-family: inherit;
}