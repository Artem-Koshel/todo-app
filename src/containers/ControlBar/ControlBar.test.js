import React from 'react';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderWithProviders from '../../utils/test-utils';
import ControlBar from './ControlBar';

const mockRadio = jest.fn();
jest.mock('../../components/TextRadioButtons/TextRadioButtons', () => (props) => {
  mockRadio(props);
  return <div data-testid="text-radio-buttons" />;
});

describe('ControlBar component', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it('renders correctly', () => {
    const state = {
      todoList: [
        {
          id: 1,
          done: true,
        },
        {
          id: 2,
          done: false,
        },
        {
          id: 3,
          done: true,
        },
      ],
      selectedFilter: 'All',
    };
    const { container } = renderWithProviders(<ControlBar />, { preloadedState: state });

    expect(container).toMatchSnapshot();
    expect(mockRadio).toBeCalledWith(
      expect.objectContaining({
        name: 'filters',
        selectedValue: 'All',
        values: ['All', 'Active', 'Completed'],
        onChange: expect.any(Function),
      })
    );
  });
});
