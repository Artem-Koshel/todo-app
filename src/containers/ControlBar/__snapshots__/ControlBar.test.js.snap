// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlBar component renders correctly 1`] = `
<div>
  <div
    class="controlBar"
  >
    <span
      class="filters"
    >
      1 items left
    </span>
    <div
      data-testid="text-radio-buttons"
    />
    <button
      class="clearCompleted"
      type="button"
    >
      Clear completed
    </button>
  </div>
</div>
`;
